core/internal/cltest/factories.go

```diff
@@ -510,3 +510,12 @@ func BuildTaskRequests(t *testing.T, initrs []models.TaskSpec) []models.TaskSpec
 	require.NoError(t, err)
 	return dst
 }
+
+// MockNewLinkEarned mocks a link earning
+func MockNewLinkEarned(jobID string, rewardAmt *assets.Link) models.LinkEarned {
```

Comment:
As a team, we've had some miscommunication in the past about what is or isn't a mock.

> * Dummy objects are passed around but never actually used. Usually they are just used to fill parameter lists.
> * Fake objects actually have working implementations, but usually take some shortcut which makes them not suitable for production (an InMemoryTestDatabase is a good example).
> * Stubs provide canned answers to calls made during the test, usually not responding at all to anything outside what's programmed in for the test.
> * Spies are stubs that also record some information based on how they were called. One form of this might be an email service that records how many messages it was sent.
> * Mocks are pre-programmed with expectations which form a specification of the calls they are expected to receive. They can throw an exception if they receive a call they don't expect and are checked during verification to ensure they got all the calls they were expecting.
> 
> From https://martinfowler.com/bliki/TestDouble.html

Since this does not make any assertions, let's simply rename it to `FakeNewLinkEarned`.